### DISCUSSION TITLE
How to use app.storage.tab

### DISCUSSION THREAD

**Discussion by aljosahafner** [zauberzeug/nicegui#3052] [Q&A]
### Question

Hey,

So far I have been ignorant about the storage method, but as my web application grew, the users complain often that the connection is broken and the state of their app goes back to the default. What would be a way to implement `app.storage.tab`? The thing is that I don't really properly understand how to instruct the website to use this option.

Thanks in advance for any answers and tips.
Cheers!

**Comment by aljosahafner** [comment]
Here is a very very summarized mock-up of my site:
```py
@ui.page('/')
def index() -> None:
    logic = WebGUI() # all the logic is contained in this big object
    in here there is the UI using with ui.elements statements and variables are stored in logic.var1, etc.

ui.run(storage_secret='bbb', binding_refresh_interval=0.3, reconnect_timeout=20.0)
```

**Comment by falkoschindler** [comment]
Hi @aljosahafner,

Have you read https://nicegui.io/documentation/storage, in particular [this](https://nicegui.io/documentation/storage#storing_ui_state) demo? It shows how to bind against tab storage to preserve user content between tab reloads.

**Comment by aljosahafner** [comment]
Hey, thanks a lot! So you think I could get away by storing my WebGUI object into the browser tab without major modifications to my code?

**Comment by rodja** [comment]
I would rather store the state of the web gui instead of the elements itself. Also, I wonder why you experience so many disconnects. Maybe you have not utilized async/await everywhere? Please read https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-long-running-function-blocking-ui-updates

**Comment by aljosahafner** [comment]
Hey, thanks a lot for the answer. I successfully used the following:
```
@ui.page('/')
def index() -> None:
    await ui.context.client.connected()
    if 'WebGUI' not in app.storage.tab:
        app.storage.tab['WebGUI'] = WebGUI()
    then I am calling all the vars instead of logic.var1 as app.storage.tab['WebGUI'].var1 and it works great!

ui.run(storage_secret='bbb', binding_refresh_interval=0.3, reconnect_timeout=20.0)
```
